adcm_aio_client/core/config/types.py

```diff
@@ -0,0 +1,277 @@
+    group[level_name] = func(group[level_name])
+    return group[level_name]
+
+
+def get_nested_config_value(config: dict[str, Any], level_names: LevelNames) -> Any:  # noqa: ANN401
+    group, level_name = get_group_with_value(config=config, level_names=level_names)
+    return group[level_name]
+
+
+def get_group_with_value(config: dict[str, Any], level_names: LevelNames) -> tuple[dict[str, Any], ParameterName]:
+    return _get_group_with_value(config=config, level_names=level_names)
+
+
+def _get_group_with_value(
+    config: dict[str, Any], level_names: Iterable[ParameterName]
+) -> tuple[dict[str, Any], ParameterName]:
+    level_name, *rest = level_names
+    if not rest:
+        return config, level_name
+
+    return _get_group_with_value(config=config[level_name], level_names=rest)
+
+
+def level_names_to_full_name(levels: LevelNames) -> str:
+    return ensure_full_name("/".join(levels))
+
+
+def full_name_to_level_names(full: ParameterFullName) -> tuple[ParameterName, ...]:
+    return tuple(filter(bool, full.split("/")))
+
+
+def ensure_full_name(name: str) -> str:
+    if not name.startswith(ROOT_PREFIX):
+        return f"{ROOT_PREFIX}{name}"
+
+    return name
+
+
+# External Section End
+
+
+class GenericConfigData(ABC):  # noqa: B024
+    __slots__ = ("_values", "_attributes")
+
+    def __init__(self: Self, values: dict, attributes: dict) -> None:
+        self._values = values
+        self._attributes = attributes
+
+    @property
+    def values(self: Self) -> dict:
+        return self._values
+
+    @property
+    def attributes(self: Self) -> dict:
+        return self._attributes
+
+    def get_value(self: Self, parameter: LevelNames) -> Any:  # noqa: ANN401
+        return get_nested_config_value(config=self._values, level_names=parameter)
+
+    def set_value[T](self: Self, parameter: LevelNames, value: T) -> T:
+        return set_nested_config_value(config=self._values, level_names=parameter, value=value)
+
+    def get_attribute(self: Self, parameter: LevelNames, attribute: str) -> bool:
+        full_name = level_names_to_full_name(parameter)
+        return self._attributes[full_name][attribute]
+
+    def set_attribute(self: Self, parameter: LevelNames, attribute: str, value: bool) -> bool:  # noqa: FBT001
+        full_name = level_names_to_full_name(parameter)
+        self._attributes[full_name][attribute] = value
+        return value
+
+
+class ActionConfigData(GenericConfigData):
+    __slots__ = GenericConfigData.__slots__
+
+
+class ConfigData(GenericConfigData):
+    __slots__ = ("id", "description", "_values", "_attributes")
+
+    def __init__(self: Self, id: int, description: str, values: dict, attributes: dict) -> None:  # noqa: A002
+        self.id = id
+        self.description = description
+        super().__init__(values=values, attributes=attributes)
+
+    @classmethod
+    def from_v2_response(cls: type[Self], data_in_v2_format: dict) -> Self:
```

Comment:
because argument object's layout is one of v2 (specifically object format, action's one will be added later)

it's very dependant on how it is in v2, so it's worth pointing it out in internals such as this

schema is v2 dependant, work with data format is dependent on v2 format too

it may be nice to incapsulate the differences, but without example of how it will be, I don't see it adequate